### PR TITLE
Multiple Ruclei Checks

### PR DESCRIPTION
**Pull request by jen140**
Added the following ruclei checks:
drupal_cve_2019_6340
jfrog_artifactory_cve_2019_9733
jira_cve_2019_8442
jira_cve_2019_8449
total_js_cve_2019_8903
w3_total_cache_cve_2019_6715

### CODE REVIEW

File: lib/checks/drupal_cve_2019_6340.rb [hunk 1]
```diff
@@ -0,0 +1,50 @@
+
+module Intrigue
+
+    module Issue
+      class DrupalCve20196340 < BaseIssue
+        def self.generate(instance_details={})
+        {
+          added: "2021-03-30",
+          name: "drupal_cve_2019_6340 ",
+          pretty_name: "Drupal RCE with CVE-2019-6340.",
```
Comment (shpendk) [inline lib/checks/drupal_cve_2019_6340.rb:10]: Lets write out `Remote Code Execution` and include the CVE in parenthesis. Example:
```
"Citrix ShareFile StorageZones Unauthenticated Arbitrary File Read Vulnerability (CVE-2020-8982)
```

File: lib/checks/drupal_cve_2019_6340.rb [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+          category: "vulnerability",
+          status: "confirmed",
+          description: "Some field types do not properly sanitize data from non-form sources in Drupal 8.5.x before 8.5.11 and Drupal 8.6.x before 8.6.10. This can lead to arbitrary PHP code execution in some cases. A site is only affected by this if one of the following conditions is met: The site has the Drupal 8 core RESTful Web Services (rest) module enabled and allows PATCH or POST requests, or the site has another web services module enabled, like JSON:API in Drupal 8, or Services or RESTful Web Services in Drupal 7. (Note: The Drupal 7 Services module itself does not require an update at this time, but you should apply other contributed updates associated with this advisory if Services is in use.)",
+          affected_software: [ 
+            { :vendor => "Drupal", :product => "Drupal" }
+          ],
+          references: [
+            { type: "description", uri: "https://nvd.nist.gov/vuln/detail/CVE-2019-6340" }
+          ],
+          authors: ["jen140"]
```
Comment (shpendk) [inline lib/checks/drupal_cve_2019_6340.rb:21]: include original author of poc and author of nuclei template

File: lib/checks/jfrog_artifactory_cve_2019_9733.rb [hunk 3]
```diff
@@ -0,0 +1,51 @@
+
+module Intrigue
+
+    module Issue
+      class JfrogArtifactoryCve20199733 < BaseIssue
+        def self.generate(instance_details={})
+        {
+          added: "2021-03-30",
+          name: "jfrog_artifactory_cve_2019_9733",
+          pretty_name: "Jfrog Artifactory unauthenticated admin password reset CVE-2019-9733",
```
Comment (shpendk) [inline lib/checks/jfrog_artifactory_cve_2019_9733.rb:10]: All words should begin with capital letter

File: lib/checks/jfrog_artifactory_cve_2019_9733.rb [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+          status: "confirmed",
+          description: "An issue was discovered in JFrog Artifactory 6.7.3. By default, the access-admin account is used to reset the password of the admin account in case an administrator gets locked out from the Artifactory console. This is only allowable from a connection directly from localhost, but providing a X-Forwarded-For HTTP header to the request allows an unauthenticated user to login with the default credentials of the access-admin account while bypassing the whitelist of allowed IP addresses. The access-admin account can use Artifactory's API to request authentication tokens for all users including the admin account and, in turn, assume full control of all artifacts and repositories managed by Artifactory.",
+          affected_software: [ 
+            { :vendor => "Jfrog", :product => "Artifactory" }
+          ],
+          references: [
+            { type: "description", uri: "https://nvd.nist.gov/vuln/detail/CVE-2019-9733" },
+            { type: "POC", uri: "https://packetstormsecurity.com/files/152172/JFrog-Artifactory-Administrator-Authentication-Bypass.html" }
+          ],
+          authors: ["jen140"]
```
Comment (shpendk) [inline lib/checks/jfrog_artifactory_cve_2019_9733.rb:22]: include original author of poc and author of nuclei template

File: lib/checks/jira_cve_2019_8442.rb [hunk 5]
```diff
@@ -0,0 +1,50 @@
+
+module Intrigue
+
+    module Issue
+      class JiraCve20198442 < BaseIssue
+        def self.generate(instance_details={})
+        {
+          added: "2021-03-30",
+          name: "jira_cve_2019_8442 ",
+          pretty_name: "Jira Server META-INF directory exposure with CVE-2019-8442.",
```
Comment (shpendk) [inline lib/checks/jira_cve_2019_8442.rb:10]: All words should begin with capital letter

File: lib/checks/jira_cve_2019_8442.rb [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+          severity: 1,
+          category: "vulnerability",
+          status: "confirmed",
+          description: "The CachingResourceDownloadRewriteRule class in Jira before version 7.13.4, and from version 8.0.0 before version 8.0.4, and from version 8.1.0 before version 8.1.1 allows remote attackers to access files in the Jira webroot under the META-INF directory via a lax path access check.",
+          affected_software: [ 
+            { :vendor => "Atlassian", :product => "Jira Software" }
+          ],
+          references: [
+            { type: "description", uri: "https://nvd.nist.gov/vuln/detail/CVE-2019-8442" }
```
Comment (shpendk) [inline lib/checks/jira_cve_2019_8442.rb:19]: can we find more references? There should be a description of Atlassian for this issue, and maybe we can find a poc/exploit?

File: lib/checks/jfrog_artifactory_cve_2019_9733.rb [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+          severity: 1,
+          category: "vulnerability",
+          status: "confirmed",
+          description: "An issue was discovered in JFrog Artifactory 6.7.3. By default, the access-admin account is used to reset the password of the admin account in case an administrator gets locked out from the Artifactory console. This is only allowable from a connection directly from localhost, but providing a X-Forwarded-For HTTP header to the request allows an unauthenticated user to login with the default credentials of the access-admin account while bypassing the whitelist of allowed IP addresses. The access-admin account can use Artifactory's API to request authentication tokens for all users including the admin account and, in turn, assume full control of all artifacts and repositories managed by Artifactory.",
+          affected_software: [ 
+            { :vendor => "Jfrog", :product => "Artifactory" }
+          ],
+          references: [
+            { type: "description", uri: "https://nvd.nist.gov/vuln/detail/CVE-2019-9733" },
+            { type: "POC", uri: "https://packetstormsecurity.com/files/152172/JFrog-Artifactory-Administrator-Authentication-Bypass.html" }
```
Comment (shpendk) [inline lib/checks/jfrog_artifactory_cve_2019_9733.rb:20]: change type from `POC` to `exploit`

File: lib/checks/jira_cve_2019_8449.rb [hunk 8]
```diff
@@ -0,0 +1,51 @@
+
+module Intrigue
+
+    module Issue
+      class JiraCve20198449 < BaseIssue
+        def self.generate(instance_details={})
+        {
+          added: "2021-03-30",
+          name: "jira_cve_2019_8449 ",
+          pretty_name: "Jira username enumeration vulnerability with CVE-2019-8449.",
```
Comment (shpendk) [inline lib/checks/jira_cve_2019_8449.rb:10]: All words should begin with capital letter

File: lib/checks/jira_cve_2019_8449.rb [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+          severity: 1,
```
Comment (shpendk) [inline lib/checks/jira_cve_2019_8449.rb:11]: This does not sound like a severity 1 issue to me

File: lib/checks/jira_cve_2019_8449.rb [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+          category: "vulnerability",
+          status: "confirmed",
+          description: "The /rest/api/latest/groupuserpicker resource in Jira before version 8.4.0 allows remote attackers to enumerate usernames via an information disclosure vulnerability.",
+          affected_software: [ 
+            { :vendor => "Atlassian", :product => "Jira Software" }
+          ],
+          references: [
+            { type: "description", uri: "https://nvd.nist.gov/vuln/detail/CVE-2019-8449" },
+            { type: "POC", uri: "https://packetstormsecurity.com/files/156172/Jira-8.3.4-Information-Disclosure.html" }
```
Comment (shpendk) [inline lib/checks/jira_cve_2019_8449.rb:20]: change type from POC to exploit

File: lib/checks/jira_cve_2019_8442.rb [hunk 11]
```diff
@@ -0,0 +1,50 @@
...
+          ],
+          authors: ["jen140"]
```
Comment (shpendk) [inline lib/checks/jira_cve_2019_8442.rb:21]: include original author of poc and author of nuclei template

File: lib/checks/jira_cve_2019_8449.rb [hunk 12]
```diff
@@ -0,0 +1,51 @@
...
+          ],
+          authors: ["jen140"]
```
Comment (shpendk) [inline lib/checks/jira_cve_2019_8449.rb:22]: include original author of poc and author of nuclei template

File: lib/checks/total_js_cve_2019_8903.rb [hunk 13]
```diff
@@ -0,0 +1,50 @@
...
+          category: "vulnerability",
+          status: "confirmed",
+          description: "index.js in Total.js Platform before 3.2.3 allows path traversal.",
+          affected_software: [ 
+            { :vendor => "Total.js", :product => "Total.js Platform" }
+          ],
+          references: [
+            { type: "description", uri: "https://nvd.nist.gov/vuln/detail/CVE-2019-8903" }
+          ],
+          authors: ["jen140"]
```
Comment (shpendk) [inline lib/checks/total_js_cve_2019_8903.rb:21]: include original author of poc and author of nuclei template

File: lib/checks/total_js_cve_2019_8903.rb [hunk 14]
```diff
@@ -0,0 +1,50 @@
+
+module Intrigue
+
+    module Issue
+      class TotalJSCve20198903 < BaseIssue
+        def self.generate(instance_details={})
+        {
+          added: "2021-03-30",
+          name: "total_js_cve_2019_8903 ",
+          pretty_name: "Total.js path traversal vulnerability with CVE-2019-8903.",
```
Comment (shpendk) [inline lib/checks/total_js_cve_2019_8903.rb:10]: All words should begin with capital letter and the CVE number should be in parenthesis

File: lib/checks/total_js_cve_2019_8903.rb [hunk 15]
```diff
@@ -0,0 +1,50 @@
...
+          severity: 1,
+          category: "vulnerability",
+          status: "confirmed",
+          description: "index.js in Total.js Platform before 3.2.3 allows path traversal.",
```
Comment (shpendk) [inline lib/checks/total_js_cve_2019_8903.rb:14]: Maybe we can find a longer more descriptive description?

File: lib/checks/total_js_cve_2019_8903.rb [hunk 16]
```diff
@@ -0,0 +1,50 @@
...
+          affected_software: [ 
+            { :vendor => "Total.js", :product => "Total.js Platform" }
+          ],
+          references: [
+            { type: "description", uri: "https://nvd.nist.gov/vuln/detail/CVE-2019-8903" }
```
Comment (shpendk) [inline lib/checks/total_js_cve_2019_8903.rb:19]: can we find more references? There should be a description of the vendor for this issue, and maybe we can find a poc/exploit link

File: lib/checks/w3_total_cache_cve_2019_6715.rb [hunk 17]
```diff
@@ -0,0 +1,51 @@
...
+          status: "confirmed",
+          description: "pub/sns.php in the W3 Total Cache plugin before 0.9.4 for WordPress allows remote attackers to read arbitrary files via the SubscribeURL field in SubscriptionConfirmation JSON data.",
+          affected_software: [ 
+            { :vendor => "BoldGrid", :product => "W3 Total Cache" }
+          ],
+          references: [
+            { type: "description", uri: "https://nvd.nist.gov/vuln/detail/CVE-2019-6715" },
+            { type: "POC", uri: "https://packetstormsecurity.com/files/160674/WordPress-W3-Total-Cache-0.9.3-File-Read-Directory-Traversal.html" }
+          ],
+          authors: ["jen140"]
```
Comment (shpendk) [inline lib/checks/w3_total_cache_cve_2019_6715.rb:22]: include original author of poc and author of nuclei template

File: lib/checks/w3_total_cache_cve_2019_6715.rb [hunk 18]
```diff
@@ -0,0 +1,51 @@
+
+module Intrigue
+
+    module Issue
+      class W3TotalCacheCve20196715 < BaseIssue
+        def self.generate(instance_details={})
+        {
+          added: "2021-03-30",
+          name: "w3_total_cache_cve_2019_6715 ",
+          pretty_name: "W3 Total Cache Wordpress Plugin Arbitrary file read with CVE-2019-6715.",
```
Comment (shpendk) [inline lib/checks/w3_total_cache_cve_2019_6715.rb:10]: All words should begin with capital letter and the CVE number should be in parenthesis

File: lib/checks/w3_total_cache_cve_2019_6715.rb [hunk 19]
```diff
@@ -0,0 +1,51 @@
...
+          severity: 1,
+          category: "vulnerability",
+          status: "confirmed",
+          description: "pub/sns.php in the W3 Total Cache plugin before 0.9.4 for WordPress allows remote attackers to read arbitrary files via the SubscribeURL field in SubscriptionConfirmation JSON data.",
+          affected_software: [ 
+            { :vendor => "BoldGrid", :product => "W3 Total Cache" }
+          ],
+          references: [
+            { type: "description", uri: "https://nvd.nist.gov/vuln/detail/CVE-2019-6715" },
+            { type: "POC", uri: "https://packetstormsecurity.com/files/160674/WordPress-W3-Total-Cache-0.9.3-File-Read-Directory-Traversal.html" }
```
Comment (shpendk) [inline lib/checks/w3_total_cache_cve_2019_6715.rb:20]: change type from POC to exploit

**Review by shpendk — CHANGES_REQUESTED** [review]
Need some changes mostly for consistency sake. Otherwise really good job mate!

**Comment by jen140** [comment]
Fixed everything as requested (except total.js description).

**Review by shpendk — APPROVED** [review]
Looks good now, thanks again

**Pull request merged**